source/common/http/header_map_impl.cc

```diff
@@ -178,6 +179,78 @@ template <> bool HeaderMapImpl::HeaderList::isPseudoHeader(const LowerCaseString
   return key.get().c_str()[0] == ':';
 }
 
+namespace {
+
+// Loading the threshold for lazy map once to avoid changing it while an
+// header-map object is active, and to avoid costly calls to Runtime::getInteger.
+uint32_t lazyMapMinSize() {
+  CONSTRUCT_ON_FIRST_USE(
+      uint32_t, static_cast<uint32_t>(Runtime::getInteger("envoy.http.headermap.lazy_map_min_size",
+                                                          std::numeric_limits<uint32_t>::max())));
```

Comment:
As you already discovered, this is going to be first done in `HeaderMapImplUtility::getAllHeaderMapImplInfo()` and there is no Runtime at that point, so we have to somehow late update this. This is actually really ugly since not only is runtime initialized later, but it can be continuously updated either via filesystem, RTDS, etc. Here is my concrete suggestion:
1) Confirm this can be updated at runtime without issue (I think you already confirmed this but we should add tests and comments, etc.)
2) This is more work (sorry), but we have had multiple cases in which we actually need a callback that gets notified when runtime changes. Basically, I would build a system in which you can register a callback for when the snapshot gets updated, either via RTDS, filesystem. whatever.
3) Install a callback that hits the atomic if the key is set/changed.

(2) will be a good improvement for everyone, but will also enable your use case.

I would also potentially recommend consolidating this inside `HeaderMapImplUtility` so we have a single place to audit all of this type of init logic, but up to you.

Reply:
After discussing this with Josh and Alyssa offline, I've decided to read the value during the headermap instantiation.
This increases the headermap creation benchmark time from ~15ns to ~23ns when the runtime is not initialized, and to ~88ns when it is.
As for each request/response we only create a handful of headermaps, the impact should be reduced.

That said, I can see the need for (2), and if it's ok, I would like to work on that in a different PR.
What do you think?

Reply:
Sure I think it's fine to do it on every creation.

Reply:
Done.